### PR TITLE
ISSUE-1164 Update the handle message when exceeding public asset quota limit

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/1164

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala [hunk 1]
```diff
@@ -115,6 +115,7 @@ case class PublicAssetCreationSuccess(publicAssetCreationId: PublicAssetCreation
 case class PublicAssetCreationFailure(publicAssetCreationId: PublicAssetCreationId, exception: Throwable) extends PublicAssetCreationResult {
   def asPublicAssetSetError: SetError = exception match {
     case e: PublicAssetCreationParseException => e.setError
+    case e: PublicAssetQuotaLimitExceededException => SetError.overQuota(SetError.SetErrorDescription(e.getMessage))
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala:118]: BTW where are logs in this class ?

 - > parse exceptions are lost
 - > quota exceeded are insteresting info events IMO...

**Review by chibenwa — CHANGES_REQUESTED** [review]
Exception handling misses logs

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala (hunk 1 above)
Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala:118]: > > parse exceptions are lost

it is `PublicAssetCreationParseException` and `PublicAssetException` , and already handled

**Comment by chibenwa** [comment]
> The log for invalidArguments can make noise

info/debug/trace as needed

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala [hunk 2]
```diff
@@ -115,11 +115,14 @@ case class PublicAssetCreationSuccess(publicAssetCreationId: PublicAssetCreation
 case class PublicAssetCreationFailure(publicAssetCreationId: PublicAssetCreationId, exception: Throwable) extends PublicAssetCreationResult {
   def asPublicAssetSetError: SetError = exception match {
     case e: PublicAssetCreationParseException => e.setError
+    case e: PublicAssetQuotaLimitExceededException =>
+      LOGGER.info("Has an quota limit exceed when create public asset {}", publicAssetCreationId.id.value, e)
+      SetError.overQuota(SetError.SetErrorDescription(e.getMessage))
     case e: PublicAssetException => SetError.invalidArguments(SetError.SetErrorDescription(e.getMessage))
     case e: BlobNotFoundException => SetError.invalidArguments(SetError.SetErrorDescription(e.getMessage))
     case e: IllegalArgumentException => SetError.invalidArguments(SetError.SetErrorDescription(e.getMessage))
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala:123]: log

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala [hunk 3]
```diff
@@ -115,11 +115,14 @@ case class PublicAssetCreationSuccess(publicAssetCreationId: PublicAssetCreation
 case class PublicAssetCreationFailure(publicAssetCreationId: PublicAssetCreationId, exception: Throwable) extends PublicAssetCreationResult {
   def asPublicAssetSetError: SetError = exception match {
     case e: PublicAssetCreationParseException => e.setError
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetRequest.scala:117]: log

**Comment by Arsnael** [comment]
Read it. Why is the last commit description about X-SMime-Status? And why doing an update of james sha-1?

**Comment by vttranlina** [comment]
> Read it. Why is the last commit description about X-SMime-Status? And why doing an update of james sha-1?

sr, my mistake

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**